src/main/java/com/yelp/nrtsearch/server/luceneserver/doc/LoadedDocValues.java

```diff
@@ -296,6 +296,10 @@ public GeoPoint getValue() {
           LatLng.newBuilder().setLatitude(point.getLat()).setLongitude(point.getLon()).build();
       return SearchResponse.Hit.FieldValue.newBuilder().setLatLngValue(latLon).build();
     }
+
+    public double arcDistance(double lat, double lon) {
```

Comment:
Where will this method be called from?

Reply:
ads uses this in their painless.

Reply:
It would be used in a painless script for SingleLocation doc values (Ex. `return doc['location'].arcDistance(params.lat, params.lon)`)

Reply:
you can try it on a local instance before merging

Reply:
Tested locally with painless, which worked as expected
```
    @Test
    public void testLocationArcDistance() {
        double lat = 0.0;
        double lon = 0.0;
        assertScript(String.format("return doc['lat_lon'].arcDistance(%f, %f)", lat, lon),
                1.27933529230076E7, 1.2769084708787214E7);
    }
```